core/ir/isa_regdeps/disassemble.c

```diff
@@ -0,0 +1,77 @@
+/* **********************************************************
+ * Copyright (c) 2024 Google, Inc.  All rights reserved.
+ * **********************************************************/
+
+/*
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * * Redistributions of source code must retain the above copyright notice,
+ *   this list of conditions and the following disclaimer.
+ *
+ * * Redistributions in binary form must reproduce the above copyright notice,
+ *   this list of conditions and the following disclaimer in the documentation
+ *   and/or other materials provided with the distribution.
+ *
+ * * Neither the name of VMware, Inc. nor the names of its contributors may be
+ *   used to endorse or promote products derived from this software without
+ *   specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL VMWARE, INC. OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+ * OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+ * DAMAGE.
+ */
+
+#include "../globals.h"
+#include "disassemble.h"
+
+/* DR_ISA_REGDEPS instruction encodings can be at most 16 bytes and are 4 byte aligned,
+ * hence we can have at most 4 lines of 4 bytes each.
+ */
+#define REGDEPS_BYTES_PER_LINE 4
+
+int
+print_regdeps_encoding_bytes_to_buffer(char *buf, size_t bufsz,
```

Comment:
This function does not do what its name implies: it doesn't print all the bytes!  It just print 4 bytes!  Maybe best to rename ("d_r_regdeps_print_encoding_first_line"??), and probably add a comment explaining what it's doing and what it's returning, as that is not obvious.

Update: interesting, I see this parallels print_bytes_to_buffer: that also seems confusing.

Reply:
Right, I followed `print_bytes_to_buffer()`.
Perhaps I should rename both.

Reply:
Renaming done, also for the `print_[extra_]bytes_to_buffer()` routines.
Now: `d_r_print_encoding_first_line_to_buffer()` and `d_r_print_encoding_second_line_to_buffer()`.

Reply:
SGTM